lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -439,8 +438,33 @@ def content_type
         'application/json'
       end
 
-      def success_from(response)
-        !response['errorId'] && response['status'] != 'REJECTED'
+      def success_from(action, response)
+        return false if response['errorId'] || response['error_message']
+
+        case action
+        when :authorize
+          response.dig('payment', 'statusOutput', 'isAuthorized')
+        when :capture
+          capture_status = response.dig('status') || response.dig('payment', 'status')
+          %w(CAPTURED CAPTURE_REQUESTED).include?(capture_status)
+        when :void
+          card_payment_void_response_id =
+            response.dig('cardPaymentMethodSpecificOutput', 'voidResponseId')
+
+          mobile_payment_void_response_id =
+            response.dig('mobilePaymentMethodSpecificOutput', 'voidResponseId')
+
+          if %w(00 0 8 11).include?(card_payment_void_response_id) || %w(00 0 8 11).include?(mobile_payment_void_response_id)
+            true
+          else
+            response.dig('payment', 'status') == 'CANCELLED'
+          end
```

Comment:
A few suggestions for simplification here - maybe a little less readable, but more reasonable without the if/else. The last ine is basically the conditional now.

```suggestion
        when :void
          void_response_id = response.dig('cardPaymentMethodSpecificOutput', 'voidResponseId') || response.dig('mobilePaymentMethodSpecificOutput', 'voidResponseId')
          %w(00 0 8 11).include?(void_response_id) || response.dig('payment', 'status') == 'CANCELLED'
```